### ISSUE TITLE
Navigation panels not keep up with changes

### ISSUE THREAD

**Issue by SkepticalFall** [iaincollins/icarus#53]
Nav panels not keeping up with system changes and often defaulting to a previous system. The course plotted panel keeps up. I was using Icarus on 5 different devices so I went down to two and then to just one, there was no change in the panels not keeping up or showing wrong system. With my use of Icarus, this is a program breaking fault.

**Comment by alyyasser** [comment]
i can confirm this. the terminal (most of the time) were 1 planet behind from my actual scan (FSS) results. unless i mapped a planet, it won't updated.

**Comment by alyyasser** [comment]
i found workarounds for this issue:
1. do another honking after finishing FSS will update navigation panels to the latest changes/event.
2. tapping the bottom-right button on navigation map panels in icarus will also update the navigation panels to latest changes, though im not sure whether we had to have EDMC (or similar 3rd party tools that syncing data to EDSM) running for it.
![image](https://github.com/iaincollins/icarus/assets/16476111/5164eb30-4a7b-4104-8af4-8ddd432d4ce7)

**Comment by iaincollins** [comment]
> i can confirm this. the terminal (most of the time) were 1 planet/body behind from my actual scan (FSS) results. unless i mapped a planet, it won't updated.

Thanks for adding to this with your feedback. I was actually going to close this as I can't replicate it (although I know can sometimes be flakey there, I think it doesn't handle falling back nicely when it has trouble talking to ESDM and that might be it).

Seeing as you are both seeing it I'll see if I can replicate it so I can look at addressing it.

**Comment by iaincollins** [comment]
> i would also add one more thing, if we scan (FSS) a satellite or moon that orbited a planet that we have yet scan, it won't show on icarus until we scan the planet. my suggestion would be, just add the satellite or moon with the plane replaced with a placeholder for the time being, like how the game handled it, with "Unknow Body" item.

ICARUS does still depending on folks running something that submits to ESDM. If you do, then when FSS Scan an item it does show up, but currently there is a little bit of lag as it takes a few seconds to get sent over EDDN and then show up in EDSM.

The hack of being able to tap the scan is actually intended as workaround for that issue as I've been too lazy to fix it. The behavior in ICARUS has gotten better over time but probably that aspect of map behavior is worth tracking as a separate issue.

**Comment by alyyasser** [comment]
the "refresh" feature was nice, i would love to have it on the nav/list page too, since i spend most of the time on that page, and having to go back and forth on nav/list and nav/map pages only to refresh the data is kind of, not great :)

**Comment by iaincollins** [comment]
> the "refresh" and data state indicator ("fully scanned" and not) features was nice, i would love to have it on the nav/list page too, since i spend most of the time on that page, and having to go back and forth on nav/list and nav/map pages only to refresh the data is kind of, not great :)

I agree! It's definitely on my list as annoys me too.

**Comment by iaincollins** [comment]
As an update on this issue, I've been able to observe this and I think what's going on is that sometimes EDSM.net is having "a moment" and the API takes a while to respond - or the request just times out.

The app does not handle this well and gets "stuck" waiting for it to respond, or timeout.

To resolve this, the logic in the app should be decoupled and continue to work correctly, even when it cannot contact EDSM (or while EDSM is being slow). I'll leave this issue open until it's resolved.

When I fix this, I'll also improve the automatic refresh behavior, and will add scan % indicators and an explicit refresh button to both Map and List views in the Navigation Panel.

**Comment by fci525** [comment]
Most of this thread seems to be about bodies not updating, but the original report was that the displayed current system is incorrect. I am seeing this as well, where the system it thinks I'm in is 3+ jumps ago. This may be because I am jumping with a fleet carrier? But even when undocking and flying around the system, it doesn't update.

**Comment by iaincollins** [comment]
> Most of this thread seems to be about bodies not updating, but the original report was that the displayed current system is incorrect. I am seeing this as well, where the system it thinks I'm in is 3+ jumps ago. This may be because I am jumping with a fleet carrier? But even when undocking and flying around the system, it doesn't update.

Yes, it's janky when using a Fleet Carrier. It - much more rarely - can be janky when not using a Fleet Carrier, but using a Fleet Carrier to jump it can get messed up, even though there is code that is supposed to handle this but it's evidently buggy.

I get it while exploring in a carrier too, annoyingly it only happens to me when I'm not trying to replicate it. As soon as I try to investigate it, I stop having the bug. I tried to methodically address it once and it's less worse, but it's not fixed.
